internal/router/rpcservice/rpcservice.go

```diff
@@ -168,7 +173,31 @@ func (rs *RPCService) RunReadQuery(
 	if err != nil {
 		return RunReadQueryResponse{}, fmt.Errorf("calling RunReadQuery: %v", err)
 	}
-	return RunReadQueryResponse{Result: res}, nil
+	var opts []formatter.FormatOption
```

Comment:
Cleaner code suggestion: Create a method `FormatOpts()` on `RunReadQueryRequest`, and put the logic of lines L176-189 in there. So, the code here would just be `opts, err := req.FormatOpts()`

Reply:
Nice, I like that idea. Doing it.